YPP-0023.md

```diff
@@ -0,0 +1,78 @@
+# Proposal
+Enable Yearn USDC Vaults (yvUSDC) as a collateral for USDC loans
+
+# Background
+The yvUSDC token has a $281,774,830 USD market cap and 1,991 holders. The total of value locked in Yearn vaults is over $3.8 billion. Adding yvUSDC would allow us to gauge customer interest and inform our decision on adding other Yearn vault types. Furthermore, the yvUSDC price is based on USDC and we are initially only allow borrowing against it in USDC, so that makes it a lower risk and an acceptable entry point into Yearn vault collateral.
+
+To enable yvUSDC as a collateral we will be using the [pricePerShare](https://github.com/yearn/yearn-vaults/blob/main/contracts/Vault.vy) function on the [yvUSDC token contract](https://etherscan.io/token/0x5f18c75abdae578b483e5f43f12a39cf75b973a9#readContract).  This function calculates the current share value in real time as: ` total vault assets / total token supply`.
+
+# Details
+
+The [addYVUSDC-1.ts](https://github.com/yieldprotocol/environments-v2/blob/feat/yvusdc/scripts/governance/addYearnCollateral/yvUSDC/addYVUSDC-1.ts), [addYVUSDC-2.ts](https://github.com/yieldprotocol/environments-v2/blob/feat/yvusdc/scripts/governance/addYearnCollateral/yvUSDC/addYVUSDC-2.ts)  and [addYVUSDC-3.ts](https://github.com/yieldprotocol/environments-v2/blob/feat/yvusdc/scripts/governance/addYearnCollateral/yvUSDC/addYVUSDC-3.ts) scripts  will be used, with [this input](https://github.com/yieldprotocol/environments-v2/blob/feat/yvusdc/scripts/governance/addYearnCollateral/yvUSDC/addYVUSDC.mainnet.config.ts):
+
+```
+import { readAddressMappingIfExists } from '../../../../shared/helpers'
+import { USDC, YVUSDC } from '../../../../shared/constants'
+import { YEARN } from '../../../../shared/constants'
+import { FYUSDC2112, FYUSDC2203 } from '../../../../shared/constants'
+
+export const developer = '0xC7aE076086623ecEA2450e364C838916a043F9a8'
+export const deployer = '0xC7aE076086623ecEA2450e364C838916a043F9a8'
+
+export const assets: Map<string, string> = new Map([
+  [USDC,   '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48'],
+  [YVUSDC, '0x5f18c75abdae578b483e5f43f12a39cf75b973a9'],
+])
+
+// underlying, yvToken, address
+export const yearnSources: Array<[string, string, string]> = [
+  [USDC, YVUSDC,  assets.get(YVUSDC) as string]
+]
+
+// Assets for which we will have a Join
+export const assetsToAdd: Array<[string, string]> = [
+  [YVUSDC, assets.get(YVUSDC) as string],
+]
+
+// Input data: baseId, ilkId, oracle name, ratio (1000000 == 100%), inv(ratio), line, dust, dec
+export const yearnLimits: Array<[string, string, string, number, number, number, number, number]> = [
+  [USDC, YVUSDC, YEARN, 1400000, 714000, 500000,  5000,    6],
```

Comment:
We are doing an 80% loan-to-value (125% collateralization ratio).

Reply:
Also, I think we can start with a million dollar borrowing limit at first

Reply:
👍🏼